Compiler bug when initialize Time struct with Time struct

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Ubuntu 21.04
Processor: 12 cpus, 64bit, little endian, Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
CC version: cc (Ubuntu 10.3.0-1ubuntu1) 10.3.0

getwd: /var/www/vproject
vmodules: /home/tanguy/.vmodules
vroot: /var/www/vproject/v
vexe: /var/www/vproject/v/v
vexe mtime: 2022-03-05 08:13:32
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 3ef437e

Git version: git version 2.30.2
Git vroot status: weekly.2022.07-121-g3ef437e6-dirty
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 3654d6a8-dirty

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg /var/www/vproject/test/ttime/time_expire_tick_bug.v`
```v
module main
import time

fn  main() {

		utc := time.utc()

		t := time.Time{...utc}
		println(t)
}
```

**What did you expect to see?**

No error. Just print Time struct.


**What did you see instead?**
```
==================
/tmp/v_1000/time_expire_tick_bug.16243915142864898268.tmp.c:12812: error: field name expected
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/gen/c/struct.v
+++ b/vlib/v/gen/c/struct.v
@@ -245,7 +245,7 @@ fn (mut g Gen) struct_init(node ast.StructInit) {
 						}
 					}
 				}
-				g.write(field.name)
+				g.write(c_name(field.name))
 			} else {
 				if !g.zero_struct_field(field) {
 					nr_fields--

--- a/vlib/v/tests/struct_of_time_init_with_update_test.v
+++ b/vlib/v/tests/struct_of_time_init_with_update_test.v
@@ -1,0 +1,12 @@
+import time
+
+fn test_struct_of_time_init_with_update() {
+	utc := time.utc()
+
+	t := time.Time{
+		...utc
+	}
+	println(utc)
+	println(t)
+	assert t == utc
+}

